code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -80,6 +82,11 @@
 	icon_state = open ? "barrel_open" : "barrel"
 	return ..()
 
+/obj/structure/fermenting_barrel/update_overlays()
+	. = ..()
+	if(src.renamedByPlayer || !isnull(GetComponent(/datum/component/label)))
```

Comment:
Is this use of `GetComponent` common? We try to avoid all uses of it. Can `/datum/component/label` apply a trait instead for us to check?